### PR TITLE
feat(lint): add value-without-quotes lint rule

### LINKED ISSUE #43: rule: add `value-without-quotes`
### Problem

Dotenv values containing whitespace can be ambiguous or behave differently across dotenv parsers when they are not enclosed in quotes.

Vaar should report unquoted values containing internal spaces or tabs and recommend enclosing the complete value in single or double quotes.

### Proposed rule name

`value-without-quotes`

### Bad example

```env
FOO=BAR BAZ
WELCOME_MESSAGE=Hello world
DATABASE_LABEL=Primary database
`
```

### Good example

```env
FOO="BAR BAZ"
WELCOME_MESSAGE='Hello world'
DATABASE_LABEL="Primary database"


Values without whitespace should remain valid without quotes:


FOO=BAR
PORT=3000
DEBUG=true
```

### Expected output

Given:

```env
FOO=BAR BAZ
```

Vaar should report a finding similar to:

```text
.env:1 value-without-quotes: value containing whitespace should be enclosed in quotes
```

The finding should identify:

* the affected file;
* the line number;
* the `value-without-quotes` rule ID;
* a concise explanation that the value contains unquoted whitespace.

### Evidence level

Deterministic (for exact findings proven from local files, schemas, configuration, Git state or other directly inspectable local evidence.)

### Notes

The rule should:

* Detect internal spaces and tabs in unquoted values.
* Accept values enclosed in matching single quotes.
* Accept values enclosed in matching double quotes.
* Ignore values that do not contain whitespace.
* Operate on the parsed value rather than the entire raw line.
* Avoid treating whitespace before an inline comment as part of the value when the parser recognizes it as comment syntax.
* Avoid duplicating findings already covered by `quote-character`, `space-character`, or `trailing-whitespace`.
* Support execution through:

  ```sh
  vaar lint --only=value-without-quotes
  ```

### Edge cases

The implementation should include tests for:

```env
EMPTY=
NO_SPACE=value
SINGLE_QUOTED='hello world'
DOUBLE_QUOTED="hello world"
MULTIPLE_SPACES=hello   world
TAB_VALUE=hello	world
INLINE_COMMENT=value # comment
HASH_IN_QUOTES="value # not a comment"
```

### Autofix

Automatic fixing should only be added if Vaar can quote the value without changing its meaning.

In particular, the fixer must not silently change:

* variable interpolation behavior;
* escape sequences;
* embedded quote handling;
* inline comment parsing.

If semantic preservation cannot be guaranteed, the initial rule should be reporting-only.

## Acceptance criteria

* [ ] Add the `value-without-quotes` rule to the canonical rule registry.
* [ ] Report unquoted values containing internal spaces.
* [ ] Report unquoted values containing tabs.
* [ ] Accept correctly single-quoted values.
* [ ] Accept correctly double-quoted values.
* [ ] Do not report values without whitespace.
* [ ] Do not mistake recognized inline comments for value whitespace.
* [ ] Findings include the file, line number and canonical rule ID.
* [ ] The rule works with `--only` and `--skip`.
* [ ] Unit tests cover valid, invalid and edge-case inputs.
* [ ] Add the rule to the lint rule reference documentation.
* [ ] Include bad, good and example-output sections in the rule documentation.

### PR DESCRIPTION
**Pull request by gayathri2210**
## Summary

Closes  #43

Adds the `value-without-quotes` deterministic lint rule.

### Changes

- Adds the `value-without-quotes` rule.
- Detects unquoted values containing spaces or tabs.
- Ignores quoted values.
- Uses the parsed value to avoid false positives from inline comments.
- Adds unit tests.
- Adds rule documentation.
- Registers the rule in the built-in rule set.

## Testing

- [x] go test ./...

## Summary of changes

- **New Features**
  - Added deterministic lint rule `value-without-quotes` to flag unquoted `.env` values containing spaces or tabs (no automatic fix).
  - Enabled in the default rule set, so it respects selection and suppression flags.

- **Documentation**
  - Added a new rule page and updated the rule catalog with evidence level and “Bad/Good” diagnostic examples.

- **Tests**
  - Added unit and CLI end-to-end coverage to verify reporting and correct `--only` / `--skip` behavior.

### CODE REVIEW

**Comment by gayathri2210** [comment]
@TFM-HITESH Please check added my pr

**Comment by shreyaGupta1202** [comment]
@TFM-HITESH, I am aligning value-without-quotes with lint standards and edge cases, please review

Changes I am adding on top of @gayathri2210's PR:
- add missing docs catalog entry and align rule doc with template
- add tests for multiple spaces and malformed quoted values
- avoid duplicate findings by skipping non-`QuoteNone` values
- keep the rule reporting-only to avoid unsafe semantic changes

**Comment by shreyaGupta1202** [comment]
@coderabbitai full review

File: docs/lint/rules/deterministic/value-without-quotes.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+FOO=BAR
+PORT=3000
+DEBUG=true
+```
+
+## Example output
+
+```text
+.env:1 value-without-quotes: value containing whitespace should be enclosed in quotes
+```
```
[the remark this answers is not in the document]
Comment (shreyaGupta1202) [reply docs/lint/rules/deterministic/value-without-quotes.md:49]: Updated and addressed

**Comment by shreyaGupta1202** [comment]
@coderabbitai full review

**Review by shreyaGupta1202 — APPROVED** [review]
lgtm

**Comment by shreyaGupta1202** [comment]
Thank you so much @gayathri2210  for taking the time to contribute to **Vaar**  ❤️

Every contribution helps make Vaar better.

It would mean the world to us if you could ⭐ **star the repo**. It motivates us to keep building and helps more people discover Vaar 🫶 

Stay tuned to this repo, we will be adding new issues soon!

We are so happy to see your contribution and would love to know about:

* **Any features or improvements you'd like to see?** We're always open to ideas and feedback.

Thanks again for contributing - we hope to see you around the repo again!
